robottelo/entities.py

```diff
@@ -2288,25 +2288,61 @@ class Meta(object):
         server_modes = ('sat', 'sam')
 
 
-class SyncPlan(orm.Entity):
+class SyncPlan(orm.Entity, orm.EntityReadMixin, orm.EntityDeleteMixin,
+        orm.EntityCreateMixin):
     """A representation of a Sync Plan entity."""
-    organization = entity_fields.OneToOneField('Organization', required=True)
+    enabled = entity_fields.BooleanField(required=True)
     name = entity_fields.StringField(required=True)
     # how often synchronization should run must be one of: none, hourly, daily,
     # weekly.
     interval = entity_fields.StringField(
-        choices=('none', 'hourly', 'daily', 'weekly'),
+        choices=('hourly', 'daily', 'weekly'),
         required=True,
     )
     # start datetime of synchronization
     sync_date = entity_fields.DateTimeField(required=True)
     description = entity_fields.StringField()
 
-    class Meta(object):
-        """Non-field information about this entity."""
-        api_path = 'katello/api/v2/organizations/:organization_id/sync_plans'
-        server_modes = ('sat')
+    def __init__(self, organization, **kwargs):
+        """Record ``organization_id`` and set ``self.Meta.api_path``."""
+        self.Meta.organization_id = organization
```

Comment:
My main reason for using ``organization`` instead of ``organization_id`` is so that I can call ``entities.SyncPlan`` like other existing entities that require an Org ID. But I see that for anyone reading the code ``organization`` would imply an ``entitiies.Organization`` instance and not just the ID. I'll update the code.